### PR TITLE
feat: more precise metrics

### PR DESCRIPTION
**Pull request by evan-forbes**
## Description

same as #1158 but targeted at main

### CODE REVIEW

**Comment by evan-forbes** [comment]
converted this to a draft as I don't want to spend time debugging simple merge conflict compiler stuff atm

**Comment by evan-forbes** [comment]
I'm still debugging the linter. Not sure why its just now freaking out about the mocks (which weren't changed).

File: Makefile [hunk 1]
```diff
@@ -247,7 +247,7 @@ format:
 
 lint:
 	@echo "--> Running linter"
-	@go run github.com/golangci/golangci-lint/cmd/golangci-lint run
+	@go run github.com/golangci/golangci-lint/cmd/golangci-lint@v1.55.2 run
```
Comment (rootulp) [inline Makefile:250]: Good call b/c this matches what is run in CI

https://github.com/celestiaorg/celestia-core/blob/39777b4a3aa7557b7d4b4d9401b29b4a553c8afa/.github/workflows/lint.yml#L38

**Review by rootulp — APPROVED** [review]

File: Makefile (hunk 1 above)
Comment (evan-forbes) [reply Makefile:250]: yeah, using their master branch will panic

although tbh idk why the ci linter started working again since it doesn't use this

Comment (rootulp) [reply Makefile:250]: oh I thought you did this to reproduce the CI failure locally

Comment (evan-forbes) [reply Makefile:250]: nah it was just useless without using the same version of the CI

**Comment by evan-forbes** [comment]
idk if its this PR or something else, but `TestTxMempool_ReapMaxBytesMaxGas` is flakey af in CI. passes locally everytime :sob: :sob:

**Comment by rootulp** [comment]
Agreed 😞 https://github.com/celestiaorg/celestia-core/issues/1132

File: pkg/trace/schema/mempool.go [hunk 2]
```diff
@@ -115,8 +116,18 @@ func WriteMempoolPeerState(client *trace.Client, peer p2p.ID, stateUpdate, trans
...
-// LocalTable is the tracing "measurement" (aka table) for the local mempool
-// updates, such as when a tx is added or removed.
-// TODO: actually implement the local mempool tracing
-// LocalTable = "mempool_local"
+	MempoolRejectedTable = "mempool_rejected"
+	ReasonFieldKey       = "reason"
 )
+
+// WriteMempoolRejected records why a transaction was rejected.
+func WriteMempoolRejected(client *trace.Client, reason string) {
```
Comment (staheri14) [inline pkg/trace/schema/mempool.go:124]: Isn't it necessary to trace evicted transactions in InfluxDB, similar to how they are being traced in Prometheus?

**Review by staheri14 — APPROVED** [review]
LGTM!

File: pkg/trace/schema/mempool.go (hunk 2 above)
Comment (evan-forbes) [reply pkg/trace/schema/mempool.go:124]: I believe we have another metric that already does that, so this metric was only added to complement that one.

**Review by cmwaters — APPROVED** [review]
LGTM

**Pull request merged**